Mapbox/MapboxMapsStyle/Style.swift

```diff
@@ -137,8 +150,12 @@ public class Style {
      */
     public func getLayer<T: Layer>(with layerID: String, type: T.Type) -> Result<T, LayerError> {
 
+        // Try to get layer from cache if possible
+        if let cachedLayer = layerCache[layerID] as? T {
+            return .success(cachedLayer)
```

Comment:
Presumably if a dev was using these functions as well as the lower level ones things could get out of sync?

Reply:
Yep, they could. It's kind of similar to our "CameraView".

Reply:
Got rid of layer cache for now